### PR TITLE
refactor: internalizing newBPool logic

### PR DESCRIPTION
**Pull request by wei3erHase**

### CODE REVIEW

File: src/contracts/BCoWFactory.sol
Comment (0xAustrian) [inline src/contracts/BCoWFactory.sol:1]: There is a few linting warnings

File: test/unit/BCoWFactory.t.sol
Comment (0xAustrian) [inline test/unit/BCoWFactory.t.sol:1]: So there is no way of testing it?

Comment (wei3erHase) [reply test/unit/BCoWFactory.t.sol:1]: i find that we can't because of the immutables :/

File: src/contracts/BCoWFactory.sol
Comment (wei3erHase) [reply src/contracts/BCoWFactory.sol:1]: changed the order of declaration 👍

**Review by 0xAustrian — APPROVED** [review]

**Pull request merged**